packages/svelte/src/internal/client/dom/elements/bindings/select.js

```diff
@@ -91,6 +91,28 @@ export function bind_select_value(select, get_value, update) {
 		select.__value = value;
 		mounting = false;
 	});
+
+	// If one of the options gets removed from the DOM, the value might have changed
+	effect(() => {
+		var observer = new MutationObserver(() => {
+			// @ts-ignore
+			var value = select.__value;
+			select_option(select, value, mounting);
+			/** @type {HTMLOptionElement | null} */
+			var selected_option = select.querySelector(':checked');
+			if (selected_option === null || get_option_value(selected_option) !== value) {
+				update('');
+			}
+		});
+
+		observer.observe(select, {
+			childList: true
+		});
```

Comment:
You also need `subtree: true` because of `<optgroup>` ([Example](https://svelte-5-preview-a5rk26gw7-svelte.vercel.app/#H4sIAAAAAAAACoVRQW6DMBD8ysqtBJEQJD1Sg9QH9NQjcDCwaa0aY2ETpUL8vbINDaRSe_Puzs6MZydy5gI1SYuJSNYhScmLUiQi5kvZQl9QGCQR0f04NLZDdTNwZfJSlkaggfehH5V-Q4GNwRYyeNSGGQwDdgoOz3vUbVrYQWkmEKxGkULAggi4wU6nUNjVCAL2FFQwR_fIeousHbK-ISunuaieBTP_ObMYO2tx4BdsQ-80tu1XpsIrZDlcYyd4sFs0uQUgaT0a00voZSN485lN4cHiPUdxrPxerJXgDYbHCE6HOR-wA3aiid_1PCpfEtKL3RSmfbIzTVReSuoBUHPZphcmRszuge400wOy5mPNnWn_mn2YtFfG1T7UhSF2hV93kW8o_Ecsj33MK8RT8V7mk-vTZCl_OBLLscomq663uM5o4j-1RuFOsglie8Y_YtjBtiE4vr31P2z_tkUi0vUtP3NsSWqGEedq_gbcYCVtOQMAAA==)).